src/ape/api/providers.py

```diff
@@ -1593,14 +1599,6 @@ def connect(self):
         # Register atexit handler to make sure disconnect is called for normal object lifecycle.
         atexit.register(self.disconnect)
 
-        # Register handlers to ensure atexit handlers are called when Python dies.
-        def _signal_handler(signum, frame):
```

Comment:
the atexit functions were already running it seems, this wasnt doing anything but causing problems